gramlib/grammar.ml

```diff
@@ -1320,11 +1320,15 @@ and parser_of_symbol : type s tr a.
             let al =
               try ps gstate strm__ :: al with
                 Stream.Failure ->
+                if not gstate.recover then raise Stream.Failure else
+                  let bp = LStream.count strm__ in
                   let a =
                     try parse_top_symb entry symb gstate strm__ with
                       Stream.Failure ->
                         raise (Error (symb_failed entry v sep symb))
                   in
+                  let ep = LStream.count strm__ in
+                  warn_recover_continuation bp ep strm__;
```

Comment:
Semicolons in nested lets / if-then-else are problems waiting to explode in our face at the next code refactoring. I fiercely advocate to work around the footgunny OCaml syntax by banning semicolons and relyong on `let () = foo in bar` everywhere...

Reply:
Indeed, done. Ci "green" again (I guess the plugin failure is an unmerged overlay)